doc/designs/ossl-provider-load-ex.md

```diff
@@ -21,50 +21,49 @@ OSSL_PROVIDER *OSSL_PROVIDER_load_ex(OSSL_LIB_CTX *libctx, const char *name,
                                      OSSL_PARAM params[]);
 ```
 
-intended to configure the provider in load time.
+intended to configure the provider at load time.
 
 It accepts only parameters of type `OSSL_PARAM_UTF8_STRING` because any
 provider can be initialized from the config file where the values are
 represented as strings and provider init function has to deal with it.
 
-Explicitly configured parameters can contradict the parameters named in the
+Explicitly configured parameters can differ from the parameters named in the
 configuration file. Here are the current design decisions and some possible
 future steps.
 
 Real-world cases
 ----------------
 
-Many applications use PKCS#11 API with a specific drivers. OpenSSL PKCS#11
+Many applications use PKCS#11 API with specific drivers. OpenSSL PKCS#11
 provider <https://github.com/latchset/pkcs11-provider> also provides a set of
-tweaks usable in particular situations. So there are at least several scenarios
-I have in mind:
+tweaks usable in particular situations. So there are several scenarios for which
+the new API can be used:
 
 1. Configure a provider in the config file, activate on demand
 2. Load/activate a provider run-time with parameters
 
 Current design
 --------------
 
-When the provider is loaded in the current library context and activated, the
-currently loaded provider will be returned as the result of
-`OSSL_PROVIDER_load_ex` call.
+When the provider is already loaded an activated in the current library context, the
+`OSSL_PROVIDER_load_ex` call will simply return the active provider and the extra
+parameters will be ignored.
 
-When the provider is loaded in the current library context and NOT activated,
-the parameters provided int the `OSSL_PROVIDER_load_ex` call will have the
-preference.
+In all other cases, the extra parameters provided by the `OSSL_PROVIDER_load_ex` call
+will be applied and overwrite existing parameter values from the config file.
```

Comment:
Probably it's worth mentioning that the parameters from the configuration file will be completely ignored.

Reply:
Oh, then I misunderstood. I thought the extra parameters would be merged with ones from the config file, overwriting existing value.

Reply:
No. All-or-nothing.